en/resources/tutorials/templating.md

```diff
@@ -84,43 +164,154 @@ There are {{ articles.count }} articles.
+We can render a template by calling the `response.render(_:context)` function, where context is the information the template engine will use to fill out the corresponding template files.
+
+We will start with rendering the Stencil template. Below the line:  
+
+```swift
+router.add(templateEngine: StencilTemplateEngine())
+``` 
+
+add the following: 
+```swift
+// Handle HTTP GET requests to Stencil
+router.get("/articles") { request, response, next in
+
+    // The example from https://github.com/kylef/Stencil/blob/master/README.md
+    let context = [
+       "articles": [
+         ["title": "Migrating from OCUnit to XCTest", "author": "Kyle Fuller"],
+         ["title": "Memory Management with ARC", "author": "Kyle Fuller" ]
+       ]
+    ]
+
+    try response.render("StencilExample.stencil", context: context).end()
+    next()
+}
+```
+
+So lets break down what we're doing here; 
+First we've registered a GET end point which will call a block of code (a [closure](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/Closures.html)) when a GET request is made to the server: 
+`router.get("/articles") { request, response, next in ... }`
+
+Then we create a context value, which is an array of dictionaries. The point to notice here is that the `key` for the context dictionary, i.e. the `"articles"` value matches the same value in our StencilExample.template: 
+
+{% raw %}
+```
+There are {{ articles.count }} articles.
+
+{% for article in articles %}
+  - {{ article.title }} by {{ article.author }}.
+{% endfor %}
+```
+{% endraw %}
+
+This is a requirement, the key for this dictionary HAS to match the corresponding value in your Stencil template file. This is how the template engine knows where to render the information we provide.  
+
+We then call the render function, providing the name of the template file we want to use, `"StencilExample.stencil"`in this case, and the context. The render function will then use the context to fill out the template file and serve it to a specified URL, in our case this is: `https://localhost:8080`
+
+Our `postInit()` function should look like this now: 
+```swift
+func postInit() throws {
+	// Capabilities
+	initializeMetrics(app: self)
+
+	// Endpoints
+	initializeHealthRoutes(app: self)
+	router.add(templateEngine: MustacheTemplateEngine())
+	router.add(templateEngine: StencilTemplateEngine())
+
+	router.get("/articles") { request, response, next in
+
+	    let context = [
+		"articles": [
+		    ["title": "Migrating from OCUnit to XCTest", "author": "Kyle Fuller"],
+		    ["title": "Memory Management with ARC", "author": "Kyle Fuller" ]
+		]
+	    ]
+
+	    try response.render("StencilExample.stencil", context: context).end()
+	    next()
+	}
+}
+```
+
+
+So now we're ready to see if our implemtation of the Stencil template engine works, first we need to start our server. In Xcode we can just use the shortcut `cmd + R`.  
```

Comment:
Before we run the Xcode proj we need to change the Scheme:
"In the top left corner of Xcode you should a small terminal icon with the text "KituraTemplateExample-Package" next to it. Click this icon and then click "KituraTemplateExample" from the dropdown menu."